### PR TITLE
feat: support ESLint 8.x

### LINKED ISSUE #107: Support ESLint 8.x
ESLint v8.0.0 is [released](https://eslint.org/blog/2021/10/eslint-v8.0.0-released) 🎉

It would be awesome to have official ESLint 8 support. 👊
I'm happy to help where I can of course 🙂

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
ESLint v8.0.0 is [released](https://eslint.org/blog/2021/10/eslint-v8.0.0-released) 🎉

devDependency compatibility with ESLint 8:

- [x] [`@typescript-eslint/eslint-plugin`](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin) (https://github.com/typescript-eslint/typescript-eslint/issues/3738)
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3737
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3767
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3768
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3800
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3806
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3808
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3809
  - [x] [`v5.0.0`](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.0.0)
- [x] [`@typescript-eslint/parser`](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser) (https://github.com/typescript-eslint/typescript-eslint/issues/3738)
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3737
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3767
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3768
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3800
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3806
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3808
  - [x] https://github.com/typescript-eslint/typescript-eslint/pull/3809
  - [x] [`v5.0.0`](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.0.0)
- [x] [`eslint-config-airbnb-base`](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb-base) (https://github.com/airbnb/javascript/issues/2478)
  - [x] https://github.com/airbnb/javascript/pull/2495
  - [x] [`v15.0.0`](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/CHANGELOG.md#1500--2021-11-08)
- [ ] [`eslint-config-prettier`](https://github.com/prettier/eslint-config-prettier) (https://github.com/prettier/eslint-config-prettier/issues/200)
  - [ ] https://github.com/prettier/eslint-config-prettier/pull/201
  - [x] ~Release~
- [x] [`eslint-plugin-import`](https://github.com/import-js/eslint-plugin-import) (https://github.com/import-js/eslint-plugin-import/issues/2211)
  - [x] https://github.com/import-js/eslint-plugin-import/pull/2191
  - [x] [`v2.25.0`](https://github.com/import-js/eslint-plugin-import/releases/tag/v2.25.0)
- [x] [`eslint-plugin-jest`](https://github.com/jest-community/eslint-plugin-jest) (https://github.com/jest-community/eslint-plugin-jest/issues/881)
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/891
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/893
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/898
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/899
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/940
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/941
  - [x] https://github.com/jest-community/eslint-plugin-jest/pull/942
  - [x] [`v24.4.1`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v24.4.1)
  - [x] [`v24.4.2`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v24.4.2)
  - [x] [`v25.0.0`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.0)
  - [x] [`v25.0.4`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.4)
  - [x] [`v25.0.5`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.0.5)
  - [x] [`v25.1.0`](https://github.com/jest-community/eslint-plugin-jest/releases/tag/v25.1.0)
- [x] [`eslint-plugin-prettier`](https://github.com/prettier/eslint-plugin-prettier) (https://github.com/prettier/eslint-plugin-prettier/issues/427)
  - [x] https://github.com/prettier/eslint-plugin-prettier/pull/428
  - [x] ~Release~

---

BREAKING CHANGE: Requires Node@^12.22.0 || ^14.17.0 || >=16.0.0

Closes #107

### CODE REVIEW

**Comment by benkimpel** [comment]
@MichaelDeBoey Thanks for the PR. Need any help? (I know it's still in Draft, but couldn't help taking a look.)

**Comment by MichaelDeBoey** [comment]
@benkimpel I think I'm mostly waiting on the devDeps to support ESLint 8.x now 🤔

**Comment by benkimpel** [comment]
> @benkimpel I think I'm mostly waiting on the devDeps to support ESLint 8.x now thinking

Great! :+1:

**Comment by dangreenisrael** [comment]
Hey @MichaelDeBoey, you have some linting errors

**Comment by MichaelDeBoey** [comment]
@dangreenisrael Should I just ignore `no-shadow` rule on these lines?
They're reporting that i's already there on their own line

```sh
18:19  error  'StatementType' is already declared in the upper scope on line 18 column 19  no-shadow
40:19  error  'PaddingType' is already declared in the upper scope on line 40 column 19    no-shadow
```

**Comment by dangreenisrael** [comment]
I think I'm cool with that. Will you just add a comment in the code explaining why we are making an exception to the rule?

File: package.json
Comment (Belco90) [inline package.json:38]: Sticking to ESLint v7 in the dev dependencies would unblock this PR could (not really sure where the compatibility with ESLint v8 is added/tested in this PR repo). Upgrading ESLint here is just for the local environment used to develop the project, that must not block this PR to add ESLint v8 compatibility to the package.

**Review by Belco90 — CHANGES_REQUESTED** [review]

File: src/rules/padding.ts
Comment (benkimpel) [inline src/rules/padding.ts:17]: https://github.com/typescript-eslint/typescript-eslint/issues/2484#issuecomment-687257773
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md

@dangreenisrael @MichaelDeBoey The fix for this is to update the project's linting config as described in links above. It's an erroneous no-shadow error in this case. We should have been using the typescript no-shadow rule. Sorry about that @MichaelDeBoey... probably seemed pretty weird to see it shadowing itself on the same line.

Comment (benkimpel) [reply src/rules/padding.ts:17]: (My mistake originally. Shame!)

**Review by benkimpel — APPROVED** [review]
LGTM. Thank you! :pray:

**Comment by benkimpel** [comment]
@MichaelDeBoey I want to give @dangreenisrael a chance to review, but this will probably merge tomorrow.

**Comment by benkimpel** [comment]
> @MichaelDeBoey I want to give @dangreenisrael a chance to review, but this will probably merge tomorrow.

I'm going to take Dan's [comment](https://github.com/dangreenisrael/eslint-plugin-jest-formatting/pull/108#issuecomment-965678299) as tacit approval for a merge. He'll have to trigger the release to npm, though.

**Comment by MichaelDeBoey** [comment]
@dangreenisrael Can you please trigger the release?

**Comment by dangreenisrael** [comment]
`3.1.0` has been shipped. Thanks so much for your work @MichaelDeBoey!

**Pull request merged**
